File: Android project/InstagramCloneJava2/app/src/main/java/com/mirkamol/instagramclonejava/activity/SignInActivity.java
package com.mirkamol.instagramclonejava.activity;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

import com.mirkamol.instagramclonejava.R;
/*
 *In SignUpActivity, user can login using email, password
 */
public class SignInActivity extends BaseActivity {
    private static final String TAG = SignInActivity.class.toString();
    EditText et_email;
    EditText et_password;

    @Override

    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_sign_in);

        initViews();

    }

    private void initViews() {
        et_email = findViewById(R.id.et_email);
        et_password = findViewById(R.id.et_password);

        TextView b_signin = findViewById(R.id.b_signin);
        b_signin.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                callMainActivity();
            }
        });

        TextView tv_signup = findViewById(R.id.tv_signup);
        tv_signup.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                callSignUpActivity();
            }
        });
    }

    private void callSignUpActivity() {
        Intent intent =new  Intent(this, SignUpActivity.class);
        startActivity(intent);
        finish();
    }

    private void callMainActivity() {
        Intent intent =new  Intent(this, MainActivity.class);
        startActivity(intent);
        finish();
    }
}

File: Android project/InstagramClone/app/src/main/java/com/mirkamol/instagramclone/activity/SignInActivity.kt
package com.mirkamol.instagramclone.activity

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import com.mirkamol.instagramclone.R
import com.mirkamol.instagramclone.manager.AuthHandler
import com.mirkamol.instagramclone.manager.AuthManager
import com.mirkamol.instagramclone.utils.Extensions.toast
import java.lang.Exception

/*
*In SignUpActivity, user can login using email, password
 */
class SignInActivity : BaseActivity() {
    val TAG = SignInActivity::class.java.toString()
    lateinit var et_email: EditText
    lateinit var et_password: EditText
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_sign_in)

        initViews()
    }

    private fun initViews() {
        et_email = findViewById(R.id.et_email)
        et_password = findViewById(R.id.et_password)

        val b_signin = findViewById<Button>(R.id.b_signin)
        b_signin.setOnClickListener {
            val email = et_email.text.toString().trim()
            val password = et_password.text.toString().trim()

            if (email.isNotEmpty() && password.isNotEmpty())
                firebaseSignIn(email, password)
        }

        val tv_signup = findViewById<TextView>(R.id.tv_signup)
        tv_signup.setOnClickListener {
            callSignUpActivity()
        }
    }

    private fun firebaseSignIn(email: String, password: String) {
        showLoading(this)
        AuthManager.signIn(email, password, object :AuthHandler{

            override fun onSuccess(uid: String) {
                dismissLoading()
                toast(getString(R.string.str_signin_success))
                callMainActivity(context)

            }

            override fun onError(exception: Exception?) {
                dismissLoading()
                toast(getString(R.string.str_signin_failed))
            }

        })
    }

    private fun callSignUpActivity() {
        val intent = Intent(this, SignUpActivity::class.java)
        startActivity(intent)
    }

}